### ISSUE TITLE
MinIO Helm Chart versions 8.0.1 & 8.0.2 have the apiVersion in StatefulSet missing

### ISSUE THREAD

**Issue by alpozcan** [minio/charts#61]
Hi,
The following error is observed from `kubeval` during manifest validation:

```
manifest.yaml: Missing 'apiVersion' value
```

Seems to have been introduced with https://github.com/minio/charts/pull/49.

Issue does not exist with chart version `8.0.0`.

kubeval version: 0.15.0, validation done against Kubernetes spec 1.15.7.

**Comment by alpozcan** [comment]
I did a quick-and-dirty fix here using `|default` [in this PR](https://github.com/minio/charts/pull/60), but have since realised that the proper fix would have to be done in `_helpers.tpl`, and unfortunately do not have the time implement it there myself at the moment.

**Comment by alpozcan** [comment]
Please note that the chart is rendered through `helm template`, rather than installing it directly. It might be related to the fact that the Kubernetes version information isn't available during render time.

**Comment by chris-sanders** [comment]
I noticed this today as well. This causes deployments via argocd to fail to process. Pinning to 8.0.0 fixes the issue.

**Comment by harshavardhana** [comment]
The issue was actually introduced in this PR 

```
commit d195330c465dbdb85a54253775ce8bb03515555e (HEAD)
Author: ugiraud <ulrich.giraud1@gmail.com>
Date:   Thu Oct 22 17:35:58 2020 +0200

    fix(helpers): wrong apiVersion returned for kubernetes > 1.16.0 (#52)
    
    Due to calls to deprecated .Capabilities.KubeVersion.GitVersion
    Replaced by .Capabilities.APIVersions.Has and .Capabilities.KubeVersion.Version
    
    Signed-off-by: Ulrich GIRAUD <ulrich.giraud1@gmail.com>
    
    Co-authored-by: ulrich giraud <ulrich.giraud@pole-emploi.fr>

```

**Comment by harshavardhana** [comment]
Just updated 8.0.3 can you check if fixes the problem? @alpozcan @chris-sanders

**Comment by deshetti** [comment]
@harshavardhana Does this issue have to do with the same resolution as well? https://github.com/minio/charts/issues/54

I used the 8.0.3 version to install the chart in standalone mode and it doesn't create a statefulset. It does for distributed mode.

**Comment by harshavardhana** [comment]
> @harshavardhana Does this issue have to do with the same resolution as well? #54
> 
> I used the 8.0.3 version to install the chart in standalone mode and it doesn't create a statefulset. It does for distributed mode.

Statefulset is only when you are running distributed mode @deshetti that has never changed.

**Comment by chris-sanders** [comment]
I've just updated to 8.0.3 and it's deploying fine for me now.

**Comment by alpozcan** [comment]
Thanks @ harshavardhana, I can also confirm that the issue is fixed in 8.0.3.
